offload/plugins-nextgen/level_zero/include/AsyncQueue.h

```diff
@@ -0,0 +1,52 @@
+//===--- Level Zero Target RTL Implementation -----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Async Queue wrapper for Level Zero
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef OPENMP_LIBOMPTARGET_PLUGINS_NEXTGEN_LEVEL_ZERO_ASYNCQUEUE_H
+#define OPENMP_LIBOMPTARGET_PLUGINS_NEXTGEN_LEVEL_ZERO_ASYNCQUEUE_H
+
+#include <tuple>
+
+#include "L0Memory.h"
+
+namespace llvm {
```

Comment:
nit: i think we can do 
`namespace llvm::omp::target::plugin {` but not a big deal but we already do the same in `L0Context.h`